rtl/eclrtl/rtldynfield.hpp

```diff
@@ -110,7 +110,8 @@ enum class RecordTranslationMode : byte
     Payload = 2,    // Translate all fields in datasets, and only payload fields in indexes
     AlwaysDisk = 3, // Always translate - even if wouldn't normally (e.g. csv/xml source read as binary), or crcs happen to match
     AlwaysECL = 4,  // Ignore the published format - can make sense to force no translation e.g. when field names have changed
-    Unspecified = 5
+    Unspecified = 5,
+    PayloadRemoveOnly = 6 // Allow fields to be removed from the incoming dataset, but not allow fields to be missing
```

Comment:
I would have made Unspecified be 6... these values are not persisted anywhere AFAIK

Reply:
Changed Unspecified to 6.